### PR TITLE
typo: do a proper comparison not an assignment

### PR DESCRIPTION
**Pull request by xginn8**
Change-type: patch
Signed-off-by: Matthew McGinn <matthew@balena.io>

### CODE REVIEW

File: scripts/checks.sh [hunk 1]
```diff
@@ -456,9 +456,9 @@ function check_image_corruption()
 		do
 			# TODO: the timeout here is probably too short to be effective in most cases, but let's gather
 			# data and change if necessary
-			if ! ${TIMEOUT_CMD} ${ENG} save "${i}" > /dev/null ; then
+			if ! ${TIMEOUT_CMD} ${ENG} save "${i}" > /dev/null 2>&1; then
 				# this retval indicates a timeout
-				if (( $? = 124 )); then
+				if [[ "$?" -eq 124 ]]; then
```
Comment (saintaardvark) [inline scripts/checks.sh:461]: Out of curiosity, why the change from `(( ))` to `[[ ]]`?  My understanding is the former is meant for arithmetic; is it because we're not *really* doing arithmetic here, but rather checking a return value (which just happens to be expressible as arithmetic, but doesn't really capture what we intend)?

**Comment by xginn8** [comment]
TIL that `if ! {some_cmd}` masks the retval of that command :anguished: 

```
/tmp/work 6s
❯ bash -x /tmp/work/test.sh
+ timeout 1 sleep 10
+ echo 'FAILED: 0'
FAILED: 0
+ timeout 1 sleep 10
+ retval=124
+ echo 'retval: 124'
retval: 124

/tmp/work
❯ cat /tmp/work/test.sh 
#!/bin/bash

if ! timeout 1 sleep 10; then
    echo "FAILED: $?"
else
    echo "SUCCEEDED: $?"
fi

timeout 1 sleep 10
retval=$?
echo "retval: $retval"
````
